src/main/java/cms/gov/madie/measure/services/TestCaseService.java

```diff
@@ -301,4 +313,51 @@ private HapiOperationOutcome handleJsonProcessingException() {
                 + "but outcome not able to be interpreted!")
         .build();
   }
+
+  protected String enforcePatientId(String testCaseJson, Measure measure, TestCase testCase) {
+    if (!StringUtils.isEmpty(testCaseJson)) {
+      ObjectMapper objectMapper = new ObjectMapper();
+      String modifiedjsonString = testCaseJson;
+      try {
+        JsonNode rootNode = objectMapper.readTree(testCaseJson);
+        ArrayNode allEntries = (ArrayNode) rootNode.get("entry");
+        if (allEntries != null) {
+          for (JsonNode node : allEntries) {
+            if (node.get("resource") != null
+                && node.get("resource").get("resourceType") != null
+                && node.get("resource").get("resourceType").asText().equalsIgnoreCase("Patient")) {
+              JsonNode resourceNode = node.get("resource");
+              ObjectNode o = (ObjectNode) resourceNode;
+              String patientId =
+                  CaseUtils.toCamelCase(testCase.getTitle(), true, ' ')
+                      + "-"
+                      + (!StringUtils.isBlank(testCase.getSeries())
+                          ? CaseUtils.toCamelCase(testCase.getSeries(), true, ' ') + "-"
+                          : "")
+                      + measure.getCqlLibraryName()
+                      + "-"
+                      + measure.getVersion().getMajor()
+                      + "."
+                      + measure.getVersion().getMinor()
+                      + "."
+                      + measure.getVersion().getRevisionNumber();
+              o.put("id", patientId);
+
+              ByteArrayOutputStream bout = new ByteArrayOutputStream();
+              objectMapper.writerWithDefaultPrettyPrinter().writeValue(bout, rootNode);
+              byte[] objectBytes = bout.toByteArray();
+              modifiedjsonString = new String(objectBytes);
+            }
+          }
+        }
+
+        return modifiedjsonString;
+      } catch (JsonProcessingException e) {
+        log.error("Error reading testCaseJson", e);
```

Comment:
what happens if the user saves malformed JSON since we're catching exceptions and just returning unmodified JSON? For example, if there are some missing quotes or braces, would the validation just fail earlier than this? (I think that's fine as long as we're not saving test case json that lacks unique patient IDs with no error flag set)